tests/main/test_predictions_wrapper.py

```diff
@@ -203,8 +217,30 @@ def test_prediction_wrapper_regression(self, dataset_name, housing, energy, diab
         model_predict = model.predict(X_test)
 
         model_wrapper = PredictionsModelWrapperRegression(
-            X_test, model_predict)
+            X_test, model_predict,
+            should_construct_pandas_query=should_construct_pandas_query)
 
         self.verify_predict_outputs(model, model_wrapper, X_test)
         assert not hasattr(model_wrapper, "predict_proba")
         self.verify_pickle_serialization(model, model_wrapper, X_test)
+
+    @pytest.mark.parametrize('length_test_set', [10, 100, 200, 300, 400, 500, 1000])
+    def test_prediction_wrapper_regression_perf(self, housing, length_test_set, should_construct_pandas_query):
+        dataset = housing
+
+        X_train = pd.DataFrame(data=dataset[DatasetConstants.X_TRAIN],
+                               columns=dataset[DatasetConstants.FEATURES])
+        X_test = pd.DataFrame(data=dataset[DatasetConstants.X_TEST],
+                              columns=dataset[DatasetConstants.FEATURES])
+
+        y_train = dataset[DatasetConstants.Y_TRAIN]
+        model = create_lightgbm_regressor(X_train, y_train)
+
+        model_predict = model.predict(X_test[0:length_test_set])
+
+        model_wrapper = PredictionsModelWrapperRegression(
+            X_test[0:length_test_set], model_predict, should_construct_pandas_query=should_construct_pandas_query)
+
+        self.verify_predict_outputs(model, model_wrapper, X_test[0:length_test_set])
+        assert not hasattr(model_wrapper, "predict_proba")
+        self.verify_pickle_serialization(model, model_wrapper, X_test[0:length_test_set])
```

Comment:
it doesn't seem like we are really validating the performance here. do we want to check against some time limit?  or what should the strategy be for a perf test?  usually we just want to track performance over time in some chart and compare it.

Reply:
The pytest should report the time taken by these tests in the logs. The verification logic does multiple predict calls which stresses the performance of the prediction wrapper. This mimics the behavior when the RAI ecosystem calls predict on the prediction wrapper. I don't want to assert on the execution time as it depends on the overall load on system. Assert on execution time will make the test flaky.

We already have charts from telemetry to see how much time we are spending inside prediction wrapper and if the performance will improve based on this optimization.

Reply:
sounds good!